mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/eventsourcing/GroupUnregistringManagerTest.java

```diff
@@ -0,0 +1,125 @@
+    private static final GenericGroup GROUP_A = new GenericGroup("a");
+    private static final GenericGroup GROUP_B = new GenericGroup("b");
+
+    static class TestUnregisterer implements UnregisterRemovedGroupsSubscriber.Unregisterer {
+        private ImmutableList.Builder<Group> unregisteredGroups = ImmutableList.builder();
+
+        @Override
+        public Publisher<Void> unregister(Group group) {
+            return Mono.fromRunnable(() -> unregisteredGroups.add(group));
+        }
+
+        ImmutableList<Group> unregisteredGroups() {
+            return unregisteredGroups.build();
+        }
+    }
+
+    GroupUnregistringManager testee;
+    TestUnregisterer unregisterer;
+
+    @BeforeEach
+    void setUp() {
+        unregisterer = new TestUnregisterer();
+        testee = new GroupUnregistringManager(new InMemoryEventStore(), unregisterer);
+    }
+
+    @Test
+    void startShouldNotUnregisterGroupsWhenNoHistory() {
+        testee.start(ImmutableSet.of(GROUP_A)).block();
+
+        assertThat(unregisterer.unregisteredGroups())
+            .isEmpty();
+    }
+
+    @Test
+    void startShouldNotUnregisterGroupsWhenNoChanges() {
+        testee.start(ImmutableSet.of(GROUP_A)).block();
+        testee.start(ImmutableSet.of(GROUP_A)).block();
+
+        assertThat(unregisterer.unregisteredGroups())
+            .isEmpty();
+    }
+
+    @Test
+    void startShouldNotUnregisterGroupsWhenAdditions() {
+        testee.start(ImmutableSet.of(GROUP_A)).block();
+        testee.start(ImmutableSet.of(GROUP_A, GROUP_B)).block();
+
+        assertThat(unregisterer.unregisteredGroups())
+            .isEmpty();
+    }
+
+    @Test
+    void startShouldUnregisterGroupsWhenRemoval() {
+        testee.start(ImmutableSet.of(GROUP_A, GROUP_B)).block();
+        testee.start(ImmutableSet.of(GROUP_A)).block();
+
+        assertThat(unregisterer.unregisteredGroups())
+            .containsExactly(GROUP_B);
+    }
+
+    @Test
+    void startShouldUnregisterGroupsWhenSwap() {
+        testee.start(ImmutableSet.of(GROUP_B)).block();
+        testee.start(ImmutableSet.of(GROUP_A)).block();
+
+        assertThat(unregisterer.unregisteredGroups())
+            .containsExactly(GROUP_B);
+    }
+
+    @Test
+    void startShouldBeAbleToUnregisterPreviouslyUnregisteredGroups() {
+        testee.start(ImmutableSet.of(GROUP_A, GROUP_B)).block();
+        testee.start(ImmutableSet.of(GROUP_A)).block();
+        testee.start(ImmutableSet.of(GROUP_A, GROUP_B)).block();
+        testee.start(ImmutableSet.of(GROUP_A)).block();
+
+        assertThat(unregisterer.unregisteredGroups())
+            .containsExactly(GROUP_B, GROUP_B);
+    }
+
+    @Test
+    void startWithNoGroupsShouldUnregisterAllPreviousGroups() {
+        testee.start(ImmutableSet.of(GROUP_A, GROUP_B)).block();
+        testee.start(ImmutableSet.of()).block();
+
+        assertThat(unregisterer.unregisteredGroups())
+            .containsExactly(GROUP_A, GROUP_B);
+    }
+}
```

Comment:
What about the test :
```
testee.start(ImmutableSet.of(GROUP_A, GROUP_B)).block();
testee.start(ImmutableSet.of(GROUP_A)).block();
testee.start(ImmutableSet.of(GROUP_A, GROUP_B)).block();
```
???

Reply:
GroupB would be unregistered once, and re-added in the aggregate.

I don't see the value of this test.